lib/request.ts

```diff
@@ -135,15 +185,17 @@ export function getRequest({
 			}
 		}, initialPromise);
 
-	var interceptResponseOrError = async function (initialPromise) {
-		interceptors = exports.interceptors.slice().reverse();
-		return interceptors.reduce(function (promise, { response, responseError }) {
-			if (response != null || responseError != null) {
-				return promise.then(response, responseError);
-			} else {
-				return promise;
-			}
-		}, initialPromise);
+	const interceptResponseOrError = async function (initialPromise) {
+		return exports.interceptors
+			.slice()
+			.reverse()
```

Comment:
I think it'd be worth thinking about making the exposed interface be `addInterceptor: (interceptor: Interceptor) => void` instead of `interceptions: Interceptor[]`, that way internally `addInterceptor` can do `interceptors.unshift(interceptor)` and we can avoid the need to clone/reverse the array for every request and therefore reduce GC pressure

Reply:
I like the idea but this would require a breaking change probably in the SDK, and we are doing major releases often enough there lately :thinking: